docs/src/main/paradox/jdbc.md

```diff
@@ -1,8 +1,246 @@
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #atLeastOnce }
+
+The offset is stored after a time window, or limited by a number of envelopes, whatever happens first.
+This window can be defined with `withSaveOffset` of the returned `AtLeastOnceProjection`.
+The default settings for the window is defined in configuration section `akka.projection.at-least-once`.
+There is a performance benefit of not storing the offset too often, but the drawback is that there can be more
+duplicates when the projection that will be processed again when the projection is restarted.
+
+The @ref:[`ShoppingCartHandler` is shown below](#handler).
+
+## groupedWithin
+
+The envelopes can be grouped before processing, which can be useful for batch updates.
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #projection-imports #grouped }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #grouped }
+
+The envelopes are grouped within a time window, or limited by a number of envelopes, whatever happens first.
+This window can be defined with `withGroup` of the returned `GroupedProjection`. The default settings for
+the window is defined in configuration section `akka.projection.grouped`.
+
+When using `groupedWithin` the handler is a @scala[`JdbcHandler[immutable.Seq[EventEnvelope[ShoppingCart.Event]]]`]@java[`JdbcHandler<List<EventEnvelope<ShoppingCart.Event>>>`].
+The @ref:[`GroupedShoppingCartHandler` is shown below](#grouped-handler).
+
+
+The offset is stored in the same transaction used for the user defined `handler`, which means exactly-once
+processing semantics if the projection is restarted from previously stored offset.
+
+## Handler
+
+It's in the @apidoc[JdbcHandler] that you implement the processing of each envelope. It's essentially a consumer function
+from `(JdbcSession, Envelope)` to @scala[`Unit`]@java[`void`]. 
+
+A handler that is consuming `ShoppingCart.Event` from `eventsByTag` can look like this:
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #handler-imports #handler }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #handler }
+
+where the `OrderRepository` is an implementation of:
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #repository }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #repository }
+
+
+Such simple handlers can also be defined as plain functions via the helper @scala[`JdbcHandler.apply`]@java[`JdbcHandler.fromFunction`] factory method.
```

Comment:
this should be after the #handler not after the #repository

Reply:
same for Slick, will change